src/AccessToken.php

```diff
@@ -360,14 +359,28 @@ private function retrieveCertsFromLocation($url, array $options = [])
                 ));
             }
 
-            return json_decode((string) file_get_contents($url), true);
+            return [
+                json_decode((string) file_get_contents($url), true),
+                $expireTime
+            ];
         }
 
         $httpHandler = $this->httpHandler;
         $response = $httpHandler(new Request('GET', $url), $options);
 
         if ($response->getStatusCode() == 200) {
-            return json_decode((string) $response->getBody(), true);
+            if ($cacheControl = $response->getHeaderLine('Cache-Control')) {
+                array_map(function ($value) use (&$expireTime) {
+                    list($key, $value) = explode('=', $value) + [null, null];
+                    if ($key == 'max-age') {
```

Comment:
Same here probably, better to trim key and value just in case?

Reply:
done!